components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/TracingReporter.java

```diff
@@ -0,0 +1,124 @@
+            this.log.trace(this.toStructuredMessage(timeStamp, span));
+        }
+    }
+
+    public void log(Instant timeStamp, SpanData span, Map<String, ?> fields) {
+        LogLevel level = LogLevel.INFO;
+        try {
+            LogLevel level0 = (LogLevel) fields.get(LogLevel.FIELD_NAME);
+            if (level0 != null) {
+                level = level0;
+                fields.remove(LogLevel.FIELD_NAME);
+            }
+        } catch (Exception exc) {
+            log.warn("fail to read value of field {}");
+        }
+        switch (level) {
+            case TRACE:
+                if (log.isTraceEnabled()) {
+                    log.trace(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            case DEBUG:
+                if (log.isDebugEnabled()) {
+                    log.debug(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            case WARN:
+                if (log.isWarnEnabled()) {
+                    log.warn(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            case ERROR:
+                if (log.isErrorEnabled()) {
+                    log.error(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            default:
+                if (log.isInfoEnabled()) {
+                    log.info(toStructuredMessage(timeStamp, span));
+                }
+        }
+    }
+    //method for stucturing the tracer log format
+    private String toStructuredMessage(Instant timeStamp, SpanData span) {
+        try {
+            StringWriter writer = new StringWriter();
+            JsonGenerator generator = this.jsonFactory.createGenerator(writer);
+            generator.writeStartObject();
+            generator.writeNumberField(TracingConstants.LATENCY, Duration.between(span.startAt, timeStamp).toMillis());
+            generator.writeStringField(TracingConstants.OPERATION_NAME, span.operationName);
+            generator.writeObjectFieldStart(TracingConstants.TAGS);
+            Iterator itr = span.tags.entrySet().iterator();
+
+            Map.Entry map;
+            Object value;
+            while (itr.hasNext()) {
+                map = (Map.Entry) itr.next();
+                value = map.getValue();
+                if (value instanceof String) {
+                    generator.writeStringField((String) map.getKey(), (String) value);
+                } else if (value instanceof Number) {
+                    generator.writeNumberField((String) map.getKey(), ((Number) value).doubleValue());
+                } else if (value instanceof Boolean) {
+                    generator.writeBooleanField((String) map.getKey(), (Boolean) value);
+                }
+            }
+            generator.writeEndObject();
+            generator.close();
+            writer.close();
+            return writer.toString();
+        } catch (Exception e) {
```

Comment:
In here as well, shall we catch the specific exceptions as in JsonException and JsonGenerationException?

Reply:
Fixed